submissions/RomanSaiko/oop/index.js

```diff
@@ -0,0 +1,64 @@
+class Inhabitant {
+    constructor ({ species, name, gender, legs, hands, saying, friends = [] }) {
+        this.species = species
+        this.name = name
+        this.gender = gender
+        this.legs = legs
+        this.hands = hands
+        this.saying = saying
+        this.friends = friends
+    }
+}
+
+class Animal extends Inhabitant {
+    constructor ({ species, name, gender, saying, friends }) {
+        super({ species, name, gender, legs: 4, hands: 0, saying, friends })
```

Comment:
Dogs and cats derive from animals in hierarchy we've got here.
Problem is that pets do not have hands at all by design.
The point of OOP and SOLID is that objects have only the properties they really need.
Imagine you will extend your world with birds, insects... Will every inhabitant have e.g. wings props?